packages/yew/src/virtual_dom/vlist.rs

```diff
@@ -38,22 +47,30 @@ impl Deref for VList {
     type Target = Vec<VNode>;
 
     fn deref(&self) -> &Self::Target {
-        &self.children
+        match self.children {
+            Some(ref m) => m,
+            None => {
+                // This is mutable because the Vec<VNode> is not Sync
+                static mut EMPTY: Vec<VNode> = Vec::new();
+                // SAFETY: The EMPTY value is always read-only
+                unsafe { &EMPTY }
+            }
+        }
     }
 }
 
 impl DerefMut for VList {
     fn deref_mut(&mut self) -> &mut Self::Target {
         self.fully_keyed = FullyKeyedState::Unknown;
-        &mut self.children
+        self.children_mut()
     }
 }
 
 impl VList {
     /// Creates a new empty [VList] instance.
     pub const fn new() -> Self {
         Self {
-            children: Vec::new(),
+            children: None,
```

Comment:
(To answer my question earlier: why do we need an option around it? It's because we want to make a const fn constructor.)

Reply:
In addition, `Rc::new()` also results in an allocation, even if the vector is empty. :)